src/a2m-v2.cpp

```diff
@@ -2596,6 +2602,12 @@ void Ca2mv2Player::update_extra_fine_effects()
     }
 }
 
+void Ca2mv2Player::set_current_order(uint8_t new_order)
+{
+    assert(new_order < 0x80);
```

Comment:
Should this be an assert?  If we play a corrupted file then line 2676 might pass an invalid value here, and a corrupted file should never trigger an assertion failure.  It would be better to perhaps print a warning but handle it gracefully as you do on the following line.

Reply:
Yes, in general, all erroneous values should reset the order index to 0. But i suspect, assert() is being removed in production code and has meaning only in debug mode.

Reply:
Yes but often asserts aren't disabled and it means you won't be able to play a corrupted file if you're running in debug mode.

Asserts are meant to catch bugs in the code, and handling a corrupted file isn't a bug, it's a feature, so in my opinion the assert doesn't belong here because you're not using it to catch a programming error in the code.

To put it another way - if you play a corrupted file and you get an assertion failure, then it should be fixed.  But what code would you change to fix the assertion failure?  None, because there's no bug.

Reply:
Okay, replaced asserts in set_current_order() and set_ins_volume() to log warnings. A agree that we should try to ignore as much errors as possible when trying corrupt file.